demo/java-demo-client/src/main/java/org/cloudfoundry/abacus/demo/DemoApplication.java

```diff
@@ -0,0 +1,87 @@
+package org.cloudfoundry.abacus.demo;
+
+import org.cloudfoundry.abacus.demo.model.*;
+import org.cloudfoundry.abacus.demo.model.AggregatedUsage.WindowName;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.ApplicationArguments;
+import org.springframework.boot.ApplicationRunner;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.web.client.RestTemplate;
+
+import java.net.URI;
+import java.util.List;
+
+@SpringBootApplication
+public class DemoApplication implements ApplicationRunner {
+
+  private static final Logger log = LoggerFactory.getLogger(DemoApplication.class);
+
+  @Value("${collector:http://localhost:9080}")
+  private String collector;
+
+  @Value("${reporting:http://localhost:9088}")
+  private String reporting;
+
+  public static void main(String... args) throws Exception {
+    SpringApplication.run(DemoApplication.class, args);
+  }
+
+  @Override
+  public void run(ApplicationArguments args) throws Exception {
+    RestTemplate restTemplate = new RestTemplate();
+
+    postUsage(restTemplate);
+
+    Thread.sleep(2000);
```

Comment:
Is there some way to maybe poll for a non-empty report and avoid a hardcoded sleep?

Reply:
Yep - we can add 3 additional REST calls. Not sure if we have to do that in the demo. We don't recommend this in the usual Abacus POST/GET flow.